datafusion/core/src/physical_plan/aggregates/mod.rs

```diff
@@ -186,13 +194,26 @@ impl AggregateExec {
 
         let schema = Arc::new(schema);
 
+        let mut alias_map: HashMap<Column, Vec<Column>> = HashMap::new();
```

Comment:
Can you explain what this code is for?  It doesn't seem correct to me as I don't understand the  circumstances under which the output of be different 🤔 

It seems like in this case the input logical plan maybe was incorrect?

Reply:
This is to deal with the case that there are alias in the group exprs, in this case we can not derive the output partitioning from the input/child directly, need to take the alias into consideration. This is similar to the `ProjectionExec`.

For example, the input has the output partitioning 'a', the `ProjectionExec` or `AggregateExec` might have alias 'a as a1', the output partitioning of the ProjectionExec or AggregateExec should be 'a1'.
`ProjectionExec` and `AggregateExec` will never change the real data distribution, but need to respect the alias.